crates/proto/src/dnssec/dnssec_dns_handle/nsec3_validation.rs

```diff
@@ -838,3 +856,24 @@ fn wildcard_based_encloser_proof<'a>(
         closest_encloser_wildcard: wildcard_encloser,
     }
 }
+
+/// Compare two labels passed by reference.
+///
+/// The definition of "Hash order" in [RFC 5155](https://www.rfc-editor.org/rfc/rfc5155#section-1.3)
+/// says that the lexical ordering of hashes is the same as the lexical ordering of base32-encoded
+/// hashes.
+fn compare_ignore_case(left: &[u8], right: &[u8]) -> Ordering {
```

Comment:
Are you aware of [`eq_ignore_ascii_case()`](https://doc.rust-lang.org/std/primitive.char.html?search=eq_ignore_ascii_case#method.eq_ignore_ascii_case)? I guess we can't use it here because we have to provide an ordering?

Also wonder whether this code would end up clearer if we extracted a type `Nsec3Hash` that holds a `&str` or `&[u8]` internally and implements comparison/ordering traits directly.

Reply:
Yeah, we need an ordering to check that an NSEC3 record covers a hash. I suppose there isn't a `cmp_ignore_ascii_case()` in the standard library because there are two choices of case folding to do. (and RFC 4034 says to fold to lowercase)

A newtype would be a good idea. Since we're dealing with both raw hashes and base32-encoded hashes in `&[u8]`s, that would help ensure we do the right comparison on the right value.